base/shell/explorer/explorer.cpp

```diff
@@ -87,6 +87,59 @@ HideMinimizedWindows(IN BOOL bHide)
 }
 #endif
 
+static BOOL
+IsExplorerSystemShell()
+{
+    HKEY hKeyWinlogon;
+    WCHAR szPath[MAX_PATH];
+    LPWSTR szExplorer;
+    BOOL bIsSystemShell = TRUE; // Assume we are the system shell by default.
+
+    if (!GetModuleFileNameW(NULL, szPath, _countof(szPath)))
+        return FALSE;
+
+    szExplorer = PathFindFileNameW(szPath);
+
+    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+        0, KEY_READ, &hKeyWinlogon) != ERROR_SUCCESS)
+    {
+        // No registry access.
+        bIsSystemShell = TRUE;
+    }
+    else
+    {
+        DWORD cbShell;
+        LSTATUS status;
+
+        // Get length of the "Shell" key
+        status = RegQueryValueExW(hKeyWinlogon, L"Shell", 0, NULL, NULL, &cbShell);
+
+        if (status == ERROR_SUCCESS)
+        {
+            LPWSTR szShell;
+            DWORD dwType;
+
+            szShell = (LPWSTR)malloc(cbShell);
```

Comment:
Generally speaking, I would prefer using `new` and `delete` rather than `malloc` and `free` since this is C++. However, this API doesn't make that very pretty.

Reply:
Fixed in latest commit